components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1112,6 +1112,13 @@ private OAuthResponse handleSuccessAuthorization(OAuthMessage oAuthMessage, OIDC
         }
         if (isResponseTypeNotIdTokenOrNone(responseType, authzRespDTO)) {
             setAccessToken(authzRespDTO, builder);
+
+            // Append scopes to the token response if the grant type is implicit
+            String[] scopes = authzRespDTO.getScope();
+            if (scopes != null && scopes.length > 0) {
+                String scopeString =  StringUtils.join(scopes, " ");
+                builder.setScope(scopeString.trim());
+            }
```

Comment:
Lets add this in a private method like `setScopes`

Reply:
Fixed